tests/exploratory_analyses/k_means_cluster_test.py

```diff
@@ -0,0 +1,44 @@
+import geopandas as gdp
+import numpy as np
+import pandas as pd
+import pytest
+
+from eis_toolkit.exceptions import EmptyDataFrameException, InvalidParameterValueException
+from eis_toolkit.exploratory_analyses.k_means_cluster import k_means_clustering
+
+df = pd.DataFrame(
+    {
+        "Location": ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J"],
+        "Latitude": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        "Longitude": [-1, -2, -3, -4, -5, -6, -7, -8, -9, -10],
+    }
+)
+gdf = gdp.GeoDataFrame(df, geometry=gdp.points_from_xy(df.Longitude, df.Latitude), crs="EPSG:4326")
+
+
+def test_k_means_clustering_output():
+    """Test that k-means function assings data points into correct clusters."""
+    kmeans_gdf = k_means_clustering(data=gdf, number_of_clusters=2, random_state=0)
+    kmeans_labels = kmeans_gdf["cluster"]
+    expected_labels = [1, 1, 1, 1, 1, 0, 0, 0, 0, 0]
+    assert list(kmeans_labels) == expected_labels
+
+
+def test_number_of_clusters():
+    """Test that number of clusters given as a parameter is non-negative and greater than zero."""
+    with pytest.raises(InvalidParameterValueException):
+        k_means_clustering(data=gdf, number_of_clusters=0)
```

Comment:
You could name this "test_invalid_number_of_clusters"

Reply:
And also modify the docstring according to the usual style of our exception tests.